### PR TITLE
feat: import auth repo and add sa key for tests

### PR DESCRIPTION
**Pull request by bharathkkb**
This imports auth repo to be managed by Terraform and sets up a SA key for use in integration tests.

### CODE REVIEW

File: gh/repos/repos.tf [hunk 1]
```diff
@@ -93,6 +93,15 @@ locals {
       description : "This action allows you to ssh into a Compute Engine instance.",
       templated : false,
     },
+    {
+      name : "auth",
+      description : "GitHub Action for authenticating to Google Cloud with GitHub Actions OIDC tokens and Workload Identity Federation.",
+      templated : false,
+      topics : ["github-actions", "google-cloud", "identity", "security"],
+      delete_branch_on_merge : true,
+      has_downloads : true,
```
Comment (bharathkkb) [inline gh/repos/repos.tf:102]: We can probably remove this. This showed up as a diff after I tf imported the repo but seems to be deprecated.

File: gh/modules/actions-repo/variables.tf [hunk 2]
```diff
@@ -80,4 +80,22 @@ variable "enforce_admins" {
...
+variable "topics" {
+  type        = list(string)
+  default     = null
+  description = "The list of topics of the repository."
+}
+
+variable "has_downloads" {
+  type        = bool
+  default     = false
+  description = " Set to true to enable the (deprecated) downloads features on the repository."
```
Comment (averikitsch) [inline gh/modules/actions-repo/variables.tf:100]: nit: extra space after quote

**Review by averikitsch — APPROVED** [review]

File: gh/modules/actions-repo/variables.tf (hunk 2 above)
Comment (bharathkkb) [reply gh/modules/actions-repo/variables.tf]: fixed

**Comment by bharathkkb** [comment]
This has been applied

**Pull request merged**